src/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -143,11 +143,23 @@ public void TimesIncludeMillisecondPart_Windows()
         public void TimesIncludeMillisecondPart_OSX()
         {
             T item = GetExistingItem();
-            Assert.All(TimeFunctions(), (function) =>
+            string driveFormat = new DriveInfo(Path.GetTempPath()).DriveFormat;
+            if (driveFormat.Equals("hfs", StringComparison.InvariantCultureIgnoreCase))
             {
-                DateTime time = function.Getter(item);
-                Assert.Equal(0, time.Millisecond);
-            });
+                Assert.All(TimeFunctions(), (function) =>
+                {
+                    DateTime time = function.Getter(item);
+                    Assert.Equal(0, time.Millisecond);
+                });
+            }
+            else
+            {
+                Assert.All(TimeFunctions(), (function) =>
+                {
+                    DateTime time = function.Getter(item);
+                    Assert.NotEqual(0, time.Millisecond);
```

Comment:
Can you explain this more?  Why are we verifying it's equal on hfs and not equal on everything else?

Reply:
You can't rely on milliseconds != 0 -- it will fail 1 in 1000 times. It needs to use the same approach as TimesIncludeMillisecondPart_Windows.

Reply:
Is it that every other file system other than hfs on macOS support milliseconds times?

If so,
a) seems like a potential reliability issue to assert NotEqual, as 0 is a valid number of milliseconds, so this test would theoretically fail 1/1000 times, which will make this very flaky in CI where the test potentially runs hundreds of times a day.
b) seems like the test should really be refactored, e.g. so that https://github.com/dotnet/corefx/pull/30763/files#diff-170a22c34cffbe331bbbc9364e76aa75L74 is used when on non-hfs and this test remains for hfs.

Reply:
What @stephentoub suggests is fine too.